File: Chapter-04-Strings-AssociativeArrays/Coin-Change-With-Object/Coin-Change-With-Object.py
# Given a number of US cents, return the optimal configuration of coins in an object

def makeChange(cents):
    result = {}
    coins = [100, 50, 25, 10, 5]

    i = 0
    while cents >= 10:
        if cents / coins[i]:
            result[coins[i]] = cents / coins[i]
            cents = cents % coins[i]
        i+=1

    if cents:
        result[1] = cents

    return result

myCoins = 123
myCoins = 120
myCoins = 110
myCoins = 1234
myCoins = 1220
print(makeChange(myCoins))

File: Chapter-04-Strings-AssociativeArrays/Coin-Change-With-Object/Coin-Change-With-Object.js
// Given a number of US cents, return the optimal configuration of coins in an object

function makeChange(cents){
  var result = {}
  var i = 0
  const coins = [100, 50, 25, 10, 5]

  while(cents >= 10){
    console.log(`cents = ${cents}`)
    if (Math.floor(cents / coins[i])){
      result[coins[i]] = Math.floor(cents / coins[i])
      cents = cents % coins[i]
    }
    i++
  }

  if (cents){
    result[1] = cents
  }

  return result
}

var myCoins = 123
myCoins = 120
myCoins = 110
myCoins = 1234
myCoins = 1220
console.log(makeChange(myCoins))
